File: API/API/repositorys/ITaskAssignmentRepository.vb
Public Interface ITaskAssignmentRepository
    Function Exists(taskId As Integer, userId As Guid) As Boolean

    Sub Add(assign As TaskAssignment)

    Function GetAssignment(taskId As Integer, userId As Guid) As TaskAssignment
    Function GetAssignedTasks(userId As Guid) As List(Of TaskAssignment)

    Sub Save()
    Sub Remove(assign As TaskAssignment)
End Interface

File: API_v2/Repositorys/IRepositorys/ITaskAssignmentRepository.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using API_v2.Models;

namespace API_v2.Repositorys.IRepositorys
{
    public interface ITaskAssignmentRepository
    {
        Task<bool> ExistsAsync(int taskId, Guid userId);
        Task<TaskAssignment?> GetAssignmentAsync(int taskId, Guid userId);
        Task<List<TaskAssignment>> GetAssignedTasksAsync(Guid userId);
        void Add(TaskAssignment assignment);
        void Remove(TaskAssignment assignment);
        Task SaveAsync();
    }
}
